packages/cmf/src/settings.js

```diff
@@ -17,28 +18,41 @@ export function attachRef(state, obj) {
 	if (props._ref) {
 		const ref = state.cmf.settings.ref[props._ref];
 		invariant(ref, `CMF/Settings: Reference '${props._ref}' not found`);
-		props = Object.assign(
-			{},
-			state.cmf.settings.ref[props._ref],
-			obj
-		);
+		props = Object.assign({}, state.cmf.settings.ref[props._ref], obj);
 		delete props._ref;
 	}
 	return props;
 }
 
 export function attachRefs(state, props) {
 	const attachedProps = attachRef(state, props);
-	Object.keys(attachedProps).forEach(
-		(key) => {
-			attachedProps[key] = attachRef(state, attachedProps[key]);
-		}
-	);
+	Object.keys(attachedProps).forEach(key => {
+		attachedProps[key] = attachRef(state, attachedProps[key]);
+	});
 	return attachedProps;
 }
 
+export function nonMemoizedMapStateToViewProps(state, ownProps, componentName, componentId) {
+	let viewProps = {};
+	let viewId = ownProps.view;
+	if (!ownProps.view && componentName && !componentId) {
+		viewId = componentName;
+	} else if (!ownProps.view && componentName && componentId) {
+		viewId = `${componentName}:${componentId}`;
+	}
+	if (viewId && state.cmf.settings.views[viewId]) {
+		viewProps = Object.assign({}, state.cmf.settings.views[viewId]);
+		viewProps = attachRefs(state, viewProps);
+	}
+	return viewProps;
+}
+
+memoize.Cache = Map;
```

Comment:
Why do you replace the default cache implementation ?

Reply:
So we can clear it when testing, so one test cache doesn't leak on another one.

Reply:
https://github.com/lodash/lodash/blob/4.17.4/lodash.js#L2047